Automattic/newspack-plugin
# Fix/gravityforms amp plus

### All Submissions:






### Changes proposed in this Pull Request:



Closes #1371.

Some changes were made to Jetpack Instant Search code (#1486) – @miguelpeixe please take a look to see if I haven't messed it up.

### How to test the changes in this Pull Request:

1. While on `master`, install [Gravity Forms plugin](https://www.gravityforms.com/)
2. Create a form* with a single text input, visit and AMP page and verify the AMP-originating "It appears your submission was successful…" message is displayed on submission (#1371)
3. Switch to this branch, verify the AMP-originating message is not displayed after submission and instead, GF's "Thank you for contacting us…" message is displayed
4. Create a form where a field uses conditional logic:

<img width="1243" alt="image" src="https://user-images.githubusercontent.com/7383192/156390599-8a8708a4-b427-45f7-b23b-7b0fb728d3e5.png">

5. Load the page, observe that the form with conditional logic is **not displayed**
6. Install & activate GF's Polls plugin and add a poll form to the page
7. Observe that after submitting the choice in the poll, the results are unstyled
8. Enable AMP Plus (`wp config set NEWSPACK_AMP_PLUS_ENABLED true --raw`)
9. Enable AMP Plus' handling of Gravity Form (`wp config set NEWSPACK_AMP_PLUS_GRAVITYFORMS true --raw`)
10. Load the page, observe 
    1. both simple and conditional forms are loading and can be submitted 
    2. the poll form displays results with styling

\* Visit `/wp-admin/admin.php?page=gf_new_form`, create the form, then insert it in a page via GF's block

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201860761661613
  - https://app.asana.com/0/0/1201528767060186
